fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/client/BulkDataClient.java

```diff
@@ -312,11 +309,139 @@ public PollingLocationResponse status(String job) throws Exception {
+    }
+
+    /*
+     * stops the job (local or remote batch processor).
+     */
+    private Response.Status runStopOfTenantJob(String job) throws Exception {
+        // 4 - Check the State of the Job (batchStatus)
+        //      - batchStatus - STARTING, STARTED, STOPPING, STOPPED, FAILED, COMPLETED, ABANDONED
+
+        // 4 - Try to stop the PUT /ibm/api/batch/jobinstances/<job>?action=stop
+        //     A - SC_CONTENT_REDIRECT (302) GO_TO Location to PUT
+        //       - Stop requests sent to the batch REST API must be sent directly to the executor where the job is running.
+        //       - Link - https://www.ibm.com/support/knowledgecenter/SSEQTP_liberty/com.ibm.websphere.wlp.doc/ae/rwlp_batch_rest_api.html#rwlp_batch_rest_api__STOP_requests
+        //     B - SC_ACCEPTED (202) GO_TO ACCEPTED
+        Response.Status status = Response.Status.NO_CONTENT;
+        try {
+            // NOT_LOCAL - follow location 
+            String baseUrl =
+                    properties.get(BulkDataConfigUtil.BATCH_URL).replace("jobinstances", "jobexecutions") + "/" + job
+                            + "?action=stop";
+            WebTarget target = getWebTarget(baseUrl);
+
+            // The documentation says this is a PUT, however in practice it's actually a get.
+            // @see https://www.ibm.com/support/knowledgecenter/SSEQTP_liberty/com.ibm.websphere.wlp.doc/ae/rwlp_batch_rest_api.html#rwlp_batch_rest_api__http_return_codes
+            Response r = target.request().get();
+            String responseStr = r.readEntity(String.class);
+
+            if (HttpStatus.SC_MOVED_TEMPORARILY == r.getStatus()) {
+                // It's on a different server, and must be in the cluster.
+                // If not, we're going to connect to location, and try a put.
+                String location = r.getHeaderString("location");
+                if (location != null) {
+                    // No matter what, tell people we accepted the call.
+                    target = getWebTarget(location + "?action=stop");
+                    // no assignment intentionally.
+                    target.request().get();
+                }
+                // Here, we could easily respond with an Exception/OperationOutcome, however An unexpected error has 
+                // occurred while stopping/deleting the job on a batch cluster member
+                status = Response.Status.ACCEPTED;
+            } else if (HttpStatus.SC_OK == r.getStatus()) {
+                // Check if the Status is GOOD
+                JobExecutionResponse response = JobExecutionResponse.Parser.parse(responseStr);
+                if (response.getBatchStatus().contains("STOPPING")) {
+                    // Signal that we're in a stopping condition.
+                    status = Response.Status.ACCEPTED;
+                }
+                // Don't Delete, let the client flow through again and DELETE.
+            } else if (HttpStatus.SC_CONFLICT == r.getStatus()) {
```

Comment:
when this error can happen? just curious

Reply:
Oh, this happens when there are simultaneous updates to the status and the batchStatus is in conflict.

Reply:
Added a comment in the next commit.  This means the job is stopped, and no need to change the state.  This happens in simultaneous/parallel requests.